File: ts/src/resources/health.ts
import { HttpTransport } from "../http.js"

export class HealthResource {
  constructor(private readonly http: HttpTransport, private readonly base: string) {}

  async check() {
    return this.http.getFetch()(`${this.base}/`, { method: "GET" })
      .then(res => res.json())
  }
}

File: py/src/imbrace/resources/health.py
from typing import Any
from ..http import HttpTransport, AsyncHttpTransport


class HealthResource:
    """Health domain — Sync. 1 file = 1 domain."""
    def __init__(self, http: HttpTransport, base: str):
        self._http = http
        self._base = base

    def check(self) -> Any:
        return self._http.request("GET", f"{self._base}/").json()


class AsyncHealthResource:
    """Health domain — Async. 1 file = 1 domain."""
    def __init__(self, http: AsyncHttpTransport, base: str):
        self._http = http
        self._base = base

    async def check(self) -> Any:
        res = await self._http.request("GET", f"{self._base}/")
        return res.json()
